src/main/scala/intellij/haskell/psi/impl/HaskellQuasiQuoteManipulator.scala

```diff
@@ -0,0 +1,33 @@
+package intellij.haskell.psi.impl
+
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.{AbstractElementManipulator, PsiFileFactory}
+import com.intellij.util.IncorrectOperationException
+import intellij.haskell.HaskellLanguage
+import org.jetbrains.annotations.Nullable
+
+/**
+  * @author ice1000
+  */
+class HaskellQuasiQuoteManipulator extends AbstractElementManipulator[HaskellQuasiQuoteElementImpl] {
+  @Nullable
+  @throws[IncorrectOperationException]
+  override def handleContentChange(psi: HaskellQuasiQuoteElementImpl,
+                                   range: TextRange,
+                                   newContent: String): HaskellQuasiQuoteElementImpl = {
+    val oldText = psi.getText
+    val newText = oldText.substring(0, range.getStartOffset) + newContent + oldText.substring(range.getEndOffset)
+    val newElement = PsiFileFactory
+      .getInstance(psi.getProject)
+      .createFileFromText("a.hs", HaskellLanguage.Instance, newText, false, false)
+      .getLastChild
+      .getLastChild
```

Comment:
Can you log this `newElement.getText` for me? The thing it to find the psiElement parsed

Reply:
The problem is that this
```
    val newElement = PsiFileFactory
      .getInstance(psi.getProject)
      .createFileFromText("a.hs", HaskellLanguage.Instance, newText, false, false)
      .getLastChild
      .getLastChild
```
resolves to `HaskellTopDeclarationImpl`

Reply:
In this way it works:
```
    val newElement = PsiFileFactory
      .getInstance(psi.getProject)
      .createFileFromText("a.hs", HaskellLanguage.Instance, newText, false, false)
      .getLastChild
      .getLastChild
      .getLastChild
      .getLastChild
```
but very ugly solution...

Reply:
It can be replaced by a method which find the `HaskellQuasiQuoteElementImpl` in psi tree.

Reply:
Similar as here https://github.com/rikvdkleij/intellij-haskell/blob/a080c025c1e552eccf57ef27a4ddcf9c4aa727b6/src/main/scala/intellij/haskell/psi/HaskellElementFactory.scala#L32

Reply:
Cool! Can you fix this for me?

Reply:
Yep, no problem!

Reply:
In your branch or after merging?

Reply:
I will merge and then fix.

Reply:
Finally we're getting this done, after years!!